tests/tensor_search/test_utils.py

```diff
@@ -165,6 +172,55 @@ def run():
                 return True
             assert run()
 
+    def test_read_env_vars_and_defaults_ints(self):
+        """Make sure the priority order is expected
+        (environment vars > defaults else None) """
+        for key, mock_real_environ, default_vars, expected in [
+            ("SOME_INT_VAR", dict(), dict(), None),
+            ("SOME_INT_VAR", {"SOME_INT_VAR": "1234"}, dict(), 1234),
+            ("SOME_INT_VAR", {"SOME_INT_VAR": ""}, dict(), None),
+            ("SOME_INT_VAR", dict(), {"SOME_INT_VAR": "1234"}, 1234),
+            ("SOME_INT_VAR", dict(), {"SOME_INT_VAR": ""}, None),
+            ("SOME_INT_VAR", {"SOME_INT_VAR": "111"}, {"SOME_INT_VAR": "333"}, 111),
+            ("SOME_INT_VAR", dict(), {"SOME_INT_VAR": " 123 "}, 123),
+            ("SOME_INT_VAR", {"SOME_INT_VAR": " 123 "}, dict(), 123),
+
+            # OK for default vals to be ints:
+            ("SOME_VAR", {"SOME_VAR": "111"}, {"SOME_VAR": 333}, 111),
+            ("SOME_VAR", dict(), {"SOME_VAR": 1234}, 1234),
+        ]:
+            mock_default_env_vars = mock.MagicMock()
+            mock_default_env_vars.return_value = default_vars
+
+            @mock.patch("marqo.tensor_search.configs.default_env_vars", mock_default_env_vars)
+            @mock.patch("os.environ", mock_real_environ)
+            def run():
+                result = utils.read_env_vars_and_defaults_ints(var=key)
+                assert result == expected, f"Expected {expected}, got {result}"
+                return True
+
+            assert run()
+
+    def test_read_env_vars_and_defaults_ints_invalid_values(self):
```

Comment:
(A comment for future improvements). There is a common pattern of using a mocked object across a range of parameters. It's something like a `@pytest.mark.parameterized` with a `@mock.patch("foo", return_value="bar)`. It could be worth figuring out how to do so. I think it'd make alot of our tests, a) more readable, b) easier to see for which values, what tests fail.